obal/data/modules/koji_verify_tag.py

```diff
@@ -0,0 +1,64 @@
+"""
+Verify packages against a tag in Koji
+"""
+
+import os
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.obal import get_specfile_name # pylint:disable=import-error,no-name-in-module
+from ansible.module_utils.koji_wrapper import koji # pylint:disable=import-error,no-name-in-module
+
+def main():
+    """
+    Verify a tag
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            packages=dict(type='dict', required=True),
+            tag=dict(type='str', required=True),
+            directory=dict(type='str', required=True),
+            koji_executable=dict(type='str', require=False, default='koji')
+        )
+    )
+
+    packages_for_tag = set()
+
+    for (package, attributes) in module.params['packages'].items():
+        tag = None
+
+        if 'koji_tags' in attributes:
+            for koji_tag in attributes['koji_tags']:
+                if module.params['tag'] == koji_tag['name']:
+                    tag = koji_tag
+
+            if tag and 'package_base_dir' in attributes:
+                specfile = os.path.join(attributes['package_base_dir'], package, "{}.spec".format(package))
+                scl = tag.get('scl')
+                name = get_specfile_name(os.path.join(module.params['directory'], specfile), scl)
+                packages_for_tag.add(name)
+
+    try:
+        command = ['list-pkgs', '--quiet', '--tag', module.params['tag']]
+        packages_in_koji = koji(
```

Comment:
I'd still rename this
```suggestion
        koji_output = koji(
```

Reply:
I can. I just know its a common pattern to capture something and then down below is the same thing but just re-formatting it and using multiple variables to capture that seems overkill here.

Reply:
I think it's a bad pattern. In Python 3 you have type hints which allows static code analysis using tools like [mypy](http://mypy-lang.org/). That's a good practice IMHO and helps you. It also allows IDEs to show more useful suggestions. Sadly, Ansible is incompatible with mypy for now, but it's good to learn to do it the right way:

```console
$ cat test.py 
import subprocess
def test() -> list:
    output = subprocess.check_output('x', universal_newlines=True)
    output = output.split()
    return output
$ mypy test.py 
test.py:6: error: Incompatible types in assignment (expression has type "List[str]", variable has type "str")
test.py:7: error: Incompatible return value type (got "str", expected "List[Any]")
Found 2 errors in 1 file (checked 1 source file)
```

Compare this to:
```console
$ cat test2.py 
import subprocess
def test() -> list:
    output = subprocess.check_output('x', universal_newlines=True)
    result = output.split()
    return result
$ mypy test2.py 
Success: no issues found in 1 source file
```

Now one cool thing it can do:
```console
 cat test3.py 
import subprocess
from typing import List
def test() -> List[str]:
    output = subprocess.check_output('x')
    result = output.split()
    return result
$ mypy test3.py 
test3.py:8: error: Incompatible return value type (got "List[bytes]", expected "List[str]")
Found 1 error in 1 file (checked 1 source file)
```

Note how it caught the missing `universal_newlines` parameter which causes encoding issues.

Reply:
I tend to agree with ewoud here